source/Octopus.Tentacle.Tests.Integration/TentacleCommandLineTests.cs

```diff
@@ -0,0 +1,659 @@
+
+            stdout.Should().Be(
+@"Usage: Tentacle version [<options>]
+
+Where [<options>] is any of: 
+
+      --format=VALUE         The format of the output (text,json). Defaults 
+                               to text.
+
+Or one of the common options: 
+
+      --help                 Show detailed help for this command
+");
+        }
+
+        [Test]
+        [TentacleConfigurations(testCurrentVersionOnly: true)]
+        public async Task ShowConfigurationCommand(TentacleConfigurationTestCase tc)
+        {
+            await using var clientAndTentacle = await tc.CreateBuilder().Build(CancellationToken);
+            var (_, stdout, stderr) = await RunCommandWithValidExitCodeAssert(tc, $"show-configuration --instance={clientAndTentacle.RunningTentacle.InstanceName}");
+            stderr.Should().BeNullOrEmpty();
+
+            // Actually parse and query the document just like our consumer will
+            dynamic? settings = JsonConvert.DeserializeObject(stdout);
+
+            ((string)settings.Octopus.Home).Should().Be(clientAndTentacle.RunningTentacle.HomeDirectory, "the home directory should match");
+            ((string)settings.Tentacle.Deployment.ApplicationDirectory).Should().Be(clientAndTentacle.RunningTentacle.ApplicationDirectory, "the application directory should match");
+        }
+
+        [Test]
+        [TentacleConfigurations(testCurrentVersionOnly: true)]
+        public async Task ShowConfigurationCommandOnPartiallyConfiguredTentacle(TentacleConfigurationTestCase tc)
+        {
+            var instanceId = Guid.NewGuid().ToString();
+            using var temporaryDirectory = new TemporaryDirectory();
+            await RunCommandWithValidExitCodeAssert(tc, $"create-instance --instance={instanceId} --config=\"{Path.Combine(temporaryDirectory.DirectoryPath, instanceId + ".cfg")}\"");
+            var (_, stdout, stderr) = await RunCommandWithValidExitCodeAssert(tc, $"show-configuration --instance={instanceId}");
+            stderr.Should().BeNullOrEmpty();
+
+            // Actually parse and query the document just like our consumer will
+            dynamic? settings = JsonConvert.DeserializeObject(stdout);
+            ((string)settings.Octopus.Home).Should().Be(temporaryDirectory.DirectoryPath, "the home directory should match");
+        }
+
+        [Test]
+        [TentacleConfigurations(testCurrentVersionOnly: true)]
+        [NonParallelizable]
+        public async Task ShowConfigurationCommandLooksSensibleToHumans(TentacleConfigurationTestCase tc)
+        {
+            await using var clientAndTentacle = await tc.CreateBuilder().Build(CancellationToken);
+            var (_, stdout, stderr) = await RunCommandWithValidExitCodeAssert(tc, $"show-configuration --instance={clientAndTentacle.RunningTentacle.InstanceName}");
+            stderr.Should().BeNullOrEmpty();
+
+            if (tc.TentacleType == TentacleType.Polling)
+            {
+                stdout.Should().Be($@"{{
```

Comment:
I wonder if it would be safer to parse the json then verify parts of that json.

This test I think will fail should any slight formatting or re-ordering (JSON has no order for map type objects).

Reply:
It's very much a lift and shift effort from server E2E tests. There is definite room for improvements but I would suggest we do that when needed.